pagefaultgames/pokerogue
# [Bug] Pokemon info container icon bugfixes for gender and shiny icons

## What are the changes?
During PR #1754 and PR #2029, one more bug seems to have made it through with the gender and shiny/variant indicators "(+)" sometimes persisting through battles. This appears to happen if the user catches a new gender/shiny/variant, and the following pokemon doesn't have a gender/is not shiny

## Why am I doing these changes?
This is a bug introduced in PR #1754 

## What did change?
Added logic to make the gender new indicator disappear if the pokemon caught/hatched is genderless, and added logic for the shiny/variant new indicator to disappear if the pokemon caught/hatched is not shiny/a variant

### Screenshots/Videos
This is a screenshot of catching a new shiny based on the current behaviour:

![image](https://github.com/pagefaultgames/pokerogue/assets/66582645/0e244841-de15-4a73-b009-ea2a37dd658c)

Following that up with a non shiny catch:

![image](https://github.com/pagefaultgames/pokerogue/assets/66582645/cb913e32-2df4-4b6c-9e31-2dc348f9f664)

As you can see, the shiny icon indicator is still showing between waves. 

This is an updated scenario of catching a new shiny with the update:

![image](https://github.com/pagefaultgames/pokerogue/assets/66582645/5bc68bd9-0775-426f-9bef-e774c760ce31)

Followed by a non-shiny catch next turn:

![image](https://github.com/pagefaultgames/pokerogue/assets/66582645/90ac5fc1-243b-4c0a-bb0a-6f68be4786b8)

As you can see, the shiny indicator is no longer there

For gender, this is an example of catching a new pokemon with a gender in the live version:

![image](https://github.com/pagefaultgames/pokerogue/assets/66582645/9aeea895-d854-4e3f-90c4-e5010d2198a4)

Followed by catching a genderless pokemon:

![image](https://github.com/pagefaultgames/pokerogue/assets/66582645/57f0c521-b9a4-45fd-803d-530a22569ddf)

In the update, this is catching a new pokemon with a gender:

![image](https://github.com/pagefaultgames/pokerogue/assets/66582645/7b93b455-3638-4165-8e75-c41801246e48)

Followed by a genderless one:

![image](https://github.com/pagefaultgames/pokerogue/assets/66582645/8a78cfd0-2312-47ca-981b-618669b0b2a2)

## How to test the changes?
Download this branch and make a new save, and update the following overrides:

Line 41-48 - this makes it so that you have 99 masterballs when you start a run:

```
  active: true,
  pokeballs: {
    [PokeballType.POKEBALL]: 5,
    [PokeballType.GREAT_BALL]: 0,
    [PokeballType.ULTRA_BALL]: 0,
    [PokeballType.ROGUE_BALL]: 0,
    [PokeballType.MASTER_BALL]: 99,
  }
```

Line 86- this changes all the enemy pokemon to be shiny:

`export const OPP_SHINY_OVERRIDE: boolean = true;`

Once this is done, start a new run. When you're facing against the first pokemon, make sure it's a pokemon you don't already have a shiny for already. If you need to, you can use the following override on line 79:

`export const OPP_SPECIES_OVERRIDE: Species | integer = Species.WEEDLE;`

Where "WEEDLE" would be replaced by whatever you species you'd like to catch. Once you've confirmed you're facing off against a shiny variant you don't have already, quit the game, reset your overrides to default and load the game back up and continue your previous run. You should be facing the same shiny pokemon as before. Catch it with a masterball - if it's a new shiny it should show up with the shiny indicator, as below:

![image](https://github.com/pagefaultgames/pokerogue/assets/66582645/5bc68bd9-0775-426f-9bef-e774c760ce31)

Continue to the next wave and the next pokemon shouldn't be shiny. If so, go to catch the non shiny pokemon with a masterball, and there shouldn't be an icon, like below:

![image](https://github.com/pagefaultgames/pokerogue/assets/66582645/90ac5fc1-243b-4c0a-bb0a-6f68be4786b8)

To test gender, you'll need three files:

biome.ts (you'll need to replace the version with this one)
sessionData_Guest_gender_test.prsv
data_Guest_gender_test.prsv

You can download them all from this discord link: https://discord.com/channels/1125469663833370665/1176874654015684739/1250105282256830617

First, replace the biome.ts file located at src\data\biome.ts with the one downloaded above. To confirm it's the right version, open it up and go to line 117 - you should see this:

![image](https://github.com/pagefaultgames/pokerogue/assets/66582645/4352b7d3-fc91-40f6-828e-d0f00df196d7)

**NOTE:** Even though this biomes.ts talks about krabby and mew, we won't be using them for our testing.

Once that's done, open your local env and go to manage data -> import data. Note, doing this will wipe your local account, so maybe do this on an incognito window if you care about that. Import the "data_Guest_gender_test.prsv" file. Once that's done, go to manage data -> import session -> slot 1 (note, this will wipe your current slot 1 wave, see message before). Once that's done and reloaded, go to start a new game and confirm you haven't caught onix or ditto, like below: 

![image](https://github.com/pagefaultgames/pokerogue/assets/66582645/3a5b2855-5063-4604-a91f-b34ce33697b9)

Now you can open the slot 1 fight - it should be against an onix, and you can use a masterball against it. You should catch it and see that the onix is a new gender, like so:

![image](https://github.com/pagefaultgames/pokerogue/assets/66582645/9aeea895-d854-4e3f-90c4-e5010d2198a4)

The next wave should be against a ditto - catch it with a masterball and you should see that even though there's no gender, it'll still have the (+) indicator, like so:

![image](https://github.com/pagefaultgames/pokerogue/assets/66582645/57f0c521-b9a4-45fd-803d-530a22569ddf)

Now repeat the same process as above but with the new branch (including replacing the biomes.ts, importing the data file, importing the save file and confirming you haven't caught onix or ditto)

Once all that's done, go back to slot 1, catch the onix to see the below

![image](https://github.com/pagefaultgames/pokerogue/assets/66582645/7b93b455-3638-4165-8e75-c41801246e48)

And when you get to the ditto on wave 2, when you catch it you should see that the (+) indicator is no longer there for gender, like so:

![image](https://github.com/pagefaultgames/pokerogue/assets/66582645/8a78cfd0-2312-47ca-981b-618669b0b2a2)


## Checklist
